temporalio/bridge/src/runtime.rs

```diff
@@ -0,0 +1,116 @@
+use pyo3::exceptions::{PyRuntimeError, PyValueError};
+use pyo3::prelude::*;
+use std::collections::HashMap;
+use std::net::SocketAddr;
+use std::str::FromStr;
+use temporal_sdk_core::CoreRuntime;
+use temporal_sdk_core_api::telemetry::{
+    Logger, MetricsExporter, OtelCollectorOptions, TelemetryOptions, TelemetryOptionsBuilder,
+    TraceExportConfig, TraceExporter,
+};
+use url::Url;
+
+#[pyclass]
+pub struct RuntimeRef {
+    pub(crate) runtime: CoreRuntime,
+}
+
+#[derive(FromPyObject)]
+pub struct TelemetryConfig {
+    tracing: Option<TracingConfig>,
+    logging: Option<LoggingConfig>,
+    metrics: Option<MetricsConfig>,
+}
+
+#[derive(FromPyObject)]
+pub struct TracingConfig {
+    filter: String,
+    opentelemetry: OpenTelemetryConfig,
+}
+
+#[derive(FromPyObject)]
+pub struct LoggingConfig {
+    filter: String,
+    forward: bool,
+}
+
+#[derive(FromPyObject)]
+pub struct MetricsConfig {
+    opentelemetry: Option<OpenTelemetryConfig>,
+    prometheus: Option<PrometheusConfig>,
+}
```

Comment:
True, but I've been trying to avoid unions over field names down at this level (exhaustive sum types are a hint whereas field names are enforced, granted mutual exclusivity has to be enforced by me manually)

Reply:
Sum types are more than just a hint, they're _automatic_ enforcement, but either way it has to get surfaced to python a bit differently here so w/e